packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -348,7 +348,14 @@ abstract class RenderSliverScrollingPersistentHeader extends RenderSliverPersist
     }
     final double maxExtent = this.maxExtent;
     final double paintExtent = maxExtent - constraints.scrollOffset;
+    final double cacheExtent = calculateCacheOffset(
```

Comment:
I can't claim to understand the API doc for [calculateCacheOffset](https://main-api.flutter.dev/flutter/rendering/RenderSliver/calculateCacheOffset.html) (which is often used to calculate cache extents?) however this statement is troubling: `This method is not useful if there is not a 1:1 relationship between consumed scroll offset and consumed cache extent.` Presumably that's true for pinned persistent headers, so unconditionally specifying the cache extent here might not be recommended. 

@Piinks - WDYT?

Reply:
The cacheExtent here is of `RenderSliverScrollingPersistentHeader`, which is a non-pinned and non-floating variant of the sliver app bar.  So I think the consumed scroll offset and consumed cache extent is a 1:1 relationship. 

The pinned headers you mentioned are `RenderSliverPinnedPersistentHeader`, it doesn't have an a11y issue because it won't be scrolled off the screen and dropped from the semantics tree.

Reply:
@hangyujin is right, this is the scrolling render object, not the pinned one.
I think this is the second or third sliver a11y issue that was caused by the cacheExtent not being set like this. It may be worth checking the rest of the sliver library for the same issue in other slivers.